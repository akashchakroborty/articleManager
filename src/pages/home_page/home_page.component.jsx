import React from "react";
import { connect } from "react-redux";
import { HomePageContainer } from "./home_page.styles";
import ArticlesOverview from "../../components/articles-overview/articles-overview.component";
import { getArticles } from "../../redux/articles/articles.action";
import { createStructuredSelector } from "reselect";
import { selectArticles } from "../../redux/articles/articles.selectors";

export const HomePage = ({ getArticles, articles }) => {
  React.useEffect(() => {
    articles.length === 0 && getArticles();
  }, []);
  return (
    <HomePageContainer>
      <ArticlesOverview articles={articles} />
    </HomePageContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  articles: selectArticles,
});

export default connect(mapStateToProps, { getArticles })(HomePage);
