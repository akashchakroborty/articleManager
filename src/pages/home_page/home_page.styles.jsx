import styled from "styled-components";

export const HomePageContainer = styled.div`
  width: 850px;
  display: flex;
  justify-content: center;
  margin: 30px auto;
  @media screen and (max-width: 800px) {
    flex-direction: column;
    width: unset;
    align-items: center;
    margin: 20px;
  }
`;
