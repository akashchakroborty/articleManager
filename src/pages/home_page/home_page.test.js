import React from "react";
import { shallow } from "enzyme";
import { HomePage } from "./home_page.component";

describe("CreateOrEditPage component", () => {
  let wrapper;
  let mockGetArticles;
  let useEffect;
  beforeEach(() => {
    mockGetArticles = jest.fn();
    useEffect = jest.spyOn(React, "useEffect").mockImplementation((f) => f());
    const mockProps = {
      articles: [],
      getArticles: mockGetArticles,
    };

    wrapper = shallow(<HomePage {...mockProps} />);
  });

  it("should render HomePage component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
