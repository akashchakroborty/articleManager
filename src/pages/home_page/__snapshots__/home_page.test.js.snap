// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateOrEditPage component should render HomePage component 1`] = `ShallowWrapper {}`;
